fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java

```diff
@@ -221,9 +227,10 @@ public static ExecPlan createQueryPlanWithReTry(QueryStatement queryStmt,
         // only collect once to save the original olapTable info
         Set<OlapTable> olapTables = collectOriginalOlapTables(queryStmt, dbs);
         for (int i = 0; i < Config.max_query_retry_time; ++i) {
-            long planStartTime = System.nanoTime();
+            long planStartTime = OptimisticVersion.generate();
             if (!isSchemaValid) {
-                colNames = reAnalyzeStmt(queryStmt, dbs, session);
+                reAnalyzeStmt(queryStmt, dbs, session);
+                colNames = query.getColumnOutputNames();
```

Comment:
should use the `queryStmt.getQueryRelation().getColumnOutputNames()`. We need to update the colNames if the output has updated.

Reply:
resolved